src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -19,65 +19,41 @@
             [puppetlabs.puppetdb.query.report-data :as report-data]
             [puppetlabs.puppetdb.query.resources :as resources]))
 
-(defn ignore-engine-params
-  "Query engine munge functions should take two arguments, a version
-   and a list of columns to project. Some of the munge funcitons don't
-   currently adhere to that contract, so this function will wrap the
-   given function `f` and ignore those arguments"
-  [f]
-  (constantly f))
-
-(defn stream-query-result
-  "Given a query, and database connection, return a Ring response with the query
-  results.
-
-  If the query can't be parsed, a 400 is returned."
-  [entity version query paging-options db url-prefix output-fn]
-  (let [[query->sql munge-fn]
-        (case entity
+(defn entity->sql-fns
+  [entity summarize_by]
+  (case entity
           :facts [facts/query->sql facts/munge-result-rows]
-          :event-counts [event-counts/query->sql (ignore-engine-params (event-counts/munge-result-rows (first paging-options)))]
-          :aggregate-event-counts [aggregate-event-counts/query->sql (ignore-engine-params (comp (partial kitchensink/mapvals #(if (nil? %) 0 %)) first))]
+          :event-counts [event-counts/query->sql (event-counts/munge-result-rows summarize_by)]
+          :aggregate-event-counts [aggregate-event-counts/query->sql aggregate-event-counts/munge-result-rows]
           :fact-contents [fact-contents/query->sql fact-contents/munge-result-rows]
           :fact-paths [facts/fact-paths-query->sql facts/munge-path-result-rows]
           :events [events/query->sql events/munge-result-rows]
           :nodes [nodes/query->sql nodes/munge-result-rows]
-          :environments [environments/query->sql (ignore-engine-params identity)]
+          :environments [environments/query->sql (constantly identity)]
           :reports [reports/query->sql reports/munge-result-rows]
-          :report-metrics [report-data/metrics-query->sql (ignore-engine-params (report-data/munge-result-rows :metrics))]
-          :report-logs [report-data/logs-query->sql (ignore-engine-params (report-data/munge-result-rows :logs))]
+          :report-metrics [report-data/metrics-query->sql (report-data/munge-result-rows :metrics)]
+          :report-logs [report-data/logs-query->sql (report-data/munge-result-rows :logs)]
           :factsets [factsets/query->sql factsets/munge-result-rows]
           :resources [resources/query->sql resources/munge-result-rows]
           :edges [edges/query->sql edges/munge-result-rows]
-          :catalogs [catalogs/query->sql catalogs/munge-result-rows])]
+          :catalogs [catalogs/query->sql catalogs/munge-result-rows]))
 
+(defn stream-query-result
+  "Given a query, and database connection, return a Ring response with the query
+  results."
+  [entity version query paging-options db url-prefix row-callback & [count-query-promise]]
+  (let [[query->sql munge-fn] (entity->sql-fns entity (first paging-options))]
```

Comment:
I think it's a bit confusing to use (first paging-options), given that it's only applicable to event-counts